yt/yt/server/master/node_tracker_server/node_tracker.cpp

```diff
@@ -1053,18 +1055,47 @@ class TNodeTracker
             }
         }
 
-        TRack* oldNodeRack = nullptr;
+        TRack* rack = nullptr;
 
         auto* node = FindNodeByAddress(options.DefaultAddress);
         auto isNodeNew = !IsObjectAlive(node);
         if (!isNodeNew) {
             KickOutPreviousNodeIncarnation(node, options.DefaultAddress);
-            oldNodeRack = node->GetRack();
+            rack = node->GetRack();
+        }
+        if (options.Rack.has_value()) {
+            rack = FindRackByName(options.Rack.value());
+
+            TDataCenter* dataCenter = nullptr;
+            if (options.DataCenter.has_value()) {
+                if (IsObjectAlive(rack)) {
+                    if (options.DataCenter.value() != rack->GetDataCenter()->GetName()) {
+                        THROW_ERROR_EXCEPTION("Datacenter %Qv for rack %Qv differs from current datacenter %Qv", options.DataCenter.value(), rack->GetName(), rack->GetDataCenter()->GetName());
+                    }
+                }
+
+                dataCenter = FindDataCenterByName(options.DataCenter.value());
+                if (!IsObjectAlive(dataCenter)) {
+                    CreateDataCenterObject(node, options.DataCenter.value());
+                    dataCenter = FindDataCenterByName(options.DataCenter.value());
+                    if (!dataCenter) {
+                        YT_LOG_FATAL("Cant find created datacenter %Qv", options.DataCenter.value());
```

Comment:
Cant -> Cannot

Reply:
For log messages we use strict formatting pattern: https://github.com/ytsaurus/ytsaurus/blob/main/yt/styleguide/cpp.md#logging
(see the use of brackets)

Reply:
In human-readable text "datacenter" ->"Data center" here and elsewhere (e.g. on line 1073)